src/nft/types.ts

```diff
@@ -0,0 +1,230 @@
+import type { ReactNode } from 'react';
+import type { Address, Hex, TransactionReceipt } from 'viem';
+import type { Call, TransactionError } from '../transaction/types';
+
+export type ContractType = 'ERC721' | 'ERC1155';
+
+/* Lifecycle Provider */
+
+export enum LifecycleType {
+  VIEW = 'view',
+  MINT = 'mint',
+}
+
+export type NftLifecycleProviderReact = {
+  type: LifecycleType;
+  onError?: (error: NftError) => void;
+  onStatus?: (lifecycleStatus: LifecycleStatus) => void;
+  onSuccess?: (transactionReceipt?: TransactionReceipt) => void;
+  children: ReactNode;
+};
+
+export type NftLifecycleContextType = {
+  type: LifecycleType;
+  lifecycleStatus: LifecycleStatus;
+  updateLifecycleStatus: (status: LifecycleStatusUpdate) => void;
+};
+
+/* Nft Provider */
+
+export type NftContextType = {
+  contractAddress: `0x${string}`;
+  tokenId?: string;
+} & NftData;
+
+export type NftProviderReact = {
+  children: ReactNode;
+  contractAddress: `0x${string}`;
+  tokenId: string;
+  useNftData: UseNftData;
+};
+
+/* Mint Provider */
+
+export type NftPrice = {
+  amount?: number;
+  currency?: string;
+  amountUSD?: number;
+};
+
+export type NftMintContextType = {
+  quantity: number;
+  setQuantity: (quantity: string) => void;
+  buildMintTransaction: BuildMintTransaction;
+} & NftMintData;
+
+export type NftMintProviderReact = {
+  useNftMintData: UseNftMintData;
+  buildMintTransaction: BuildMintTransaction;
+  children: ReactNode;
+};
+
+type UseNftData = (contractAddress: Hex, tokenId: string) => NftData;
+
+/**
+ * Note: exported as public Type
+ */
+export type NftData = {
+  name?: string;
+  description?: string;
+  imageUrl?: string;
+  animationUrl?: string;
+  mimeType?: string;
+  ownerAddress?: `0x${string}`;
+  lastSoldPrice: NftPrice;
+  contractType?: ContractType;
+  mintDate?: Date;
+};
+
+export type UseNftMintDataProps = {
+  contractAddress: Hex;
+  tokenId?: string;
+  quantity: number;
+};
+
+type UseNftMintData = (props: UseNftMintDataProps) => NftMintData;
+
+/**
+ * Note: exported as public Type
+ */
+export type NftMintData = {
+  price?: NftPrice;
+  mintFee?: NftPrice;
+  creatorAddress?: Hex;
+  maxMintsPerWallet?: number;
+  isEligibleToMint?: boolean;
+  totalOwners?: number;
+  recentOwners?: Address[];
+  callData?: {
+    data: Hex;
+    to: Hex;
+    value: bigint;
+  }[];
+};
+
+type BuildMintTransaction = (
+  props: BuildMintTransactionDataProps,
+) => Promise<Call[]>;
```

Comment:
would be nice if this could be `Promise<Call[]> | Promise<ContractFunctionParams[]>`

Reply:
will revisit this in a later PR